File: board-example/src/main/java/hello/board/entity/Post.java
package hello.board.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@ToString(of = {"id", "title", "content"})
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class Post extends BaseEntity {

    @Getter
    @Id @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "post_id")
    private Long id;

    @Getter
    private String title;

    @Getter
    @Lob
    private String content;

    @OneToMany(mappedBy = "post", cascade = CascadeType.REMOVE, orphanRemoval = true)
    private List<Comment> commentList = new ArrayList<>();

    @Builder
    public Post(String title, String content) {
        this.title = title;
        this.content = content;
    }

    public void updateTitle(String title) {
        this.title = title;
    }

    public void updateContent(String content) {
        this.content = content;
    }

    public void addComment(Comment comment) {
        commentList.add(comment);
    }
}

File: kotlin-multi-module/domain/post/src/main/kotlin/playground/kotlinmultimodule/domain/post/Post.kt
package playground.kotlinmultimodule.domain.post

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.ManyToOne
import playground.kotlinmultimodule.domain.account.Account
import playground.kotlinmultimodule.domain.jpacommon.BaseEntity

@Entity
class Post(
    @Column
    val title: String,
    @Column
    val content: String,
    @ManyToOne(fetch = FetchType.LAZY)
    val author: Account
): BaseEntity()
